docs/0-user-guides/0-eta-user-guide/3-java-interop/0-java-interop-basics.md

```diff
@@ -118,38 +142,118 @@ As mentioned before, the `Java` monad is used to contain the implicit `this` con
+
+
+
+```eta
+main :: IO ()
+main = do
+  -- Similar to Java code:
+  -- File file = new File("./dir/prog.exe");
+  file <- java $ newFile "./dir/prog.exe"
+  putStrLn "Executing an IO action inside of Java!"
+  -- Similar to Java code:
+  -- boolean executes = file.canExecute();
+  executes <- javaWith file canExecute
+  if executes
+  then putStrLn "File can execute!"
+  else putStrLn "File cannot execute!"
+```
+
+Or:
+
+
+
+```eta
+main :: IO ()
+main = java $ do
+  -- Similar to Java code:
+  -- boolean executes = new File("./dir/prog.exe").canExecute();
+  executes <- newFile "./dir/prog.exe" >- canExecute
+  io $ putStrLn "Executing an IO action inside of Java!"
+  if executes
+  then io $ putStrLn "File can execute!"
+  else io $ putStrLn "File cannot execute!"
+```
 
 ## Java Foreign Import Declarations
 
 Foreign import declarations are used to import a Java method as an Eta monadic action, typically in the Java monad.
 
 
 
-### General Syntax
+### Simplified Syntax
+
+The full syntax of the `foreign import` declaration is quite complex, but only a subset is needed here:
 
 ```eta
-foreign import java [safety] "[import-string]" [eta-identifier]
+foreign import java unsafe "[import-string]" [eta-identifier]
   :: [arg-type-1] -> [arg-type-2] -> .. -> [return-type]
 ```
 
-1. `[safety]` - `safe` or `unsafe`.
-    - `unsafe` is the option you would typically select. In this case, the java method identified in the `[import-string]` will be run directly. This can be dangerous if the function can block in which case it will block the Eta RTS and reduce efficiency.
-    - `safe` is the option you would select for functions that you would expect to block for some time, so they will be safely run in another thread to prevent the call from blocking the Eta RTS. This option must also be used when importing a Java method that eventually calls an exported Eta function.
-
-2. `[import-string]` can take the following forms:
+1. `[import-string]` can take the following forms:
     - `[java-method-name]`: Binds to an instance method. `[java-method-name]` should be an unqualified Java instance method name.
     - `@static [java-method-name]`: Binds to a static method. `[java-method-name]` should be a fully qualified Java static method name.
     - `@new`: Binds to a constructor. The class to construct will be determined by the return type of the declaration.
     - `@field [java-field-name]`: Binds to a getter or setter of an instance field, determined by the type signature. `[java-field-name]` should be an unqualified Java instance field name.
     - `@static @field [java-field-name]`: Binds to a getter or setter of a field, determined by the type signature. `[java-field-name]` should be a fully qualified Java static field name.
-    - `@interface [java-interface-method]`: Binds to an interface method, determined by the type signature. `[java-interface-name]` should be a unqualified Java interface method name.
```

Comment:
Firstly - `@interface` has still been kept in the `General Syntax` section in the `Java Foreign Imports` page. I was getting a bit uneasy by the fact that the `Java Foreign Import Declarations` section in `Basics` was a duplicate of the `General Syntax` section in `Java Foreign Imports`. I could understand why it was so but it was also a bit confusing for me. Eventually I decided to simplify the section in `Basics` by removing the parts which a beginner wouldn't immediately need to know to start; I ended up removing `safe`/`unsafe`, `@interface`, `@wrapper` and `@wrapper @abstract` - which ended up making the section small enough for a beginner going through `Basics` to understand and start using immediately.

But I think you're right - I should have kept `@interface` in `Basics`; after all, interfaces are common in Java. Before I do that though, I've noticed a problem with this item: why does the syntax refer to `[java-interface-method]` but the description refer to `[java-interface-name]`?

Reply:
@bradrn I think it's better if it just stays in `General Syntax` rather than `Basics`. My main concern was that you hadn't re-added it anywhere. 

Re: `[java-interface-name]` that's a typo. Please feel free to fix it.  

Note that you should also add the following section which you have removed: 
- `@wrapper [java-interface-method] ` 
- `@wrapper @abstract [java-abstract-method]` and the 
- Not Present

Reply:
It's the same situation with those other three options - they aren't really essential for basic learning and usage. I'm fixing the typo right now.

Reply:
Typo fixed in latest commit.

Reply:
Just realised I didn't really address your concern that I 'hadn't re-added' those sections anywhere. If you look at file `2-java-ffi.md` everything is still there - I haven't deleted any content.

Reply:
Oops! I missed that. Ok then everything looks good. Thank you so much for the reorganization. I'll let you know when the main website has your changes.

Reply:
@bradrn Your changes are available on the main website. Please do a hard reload to see them.

Reply:
Thank you! This is an amazing project and I'm glad I could help a bit.